src/http_server/api/v1/metrics.c

```diff
@@ -211,23 +267,48 @@ void cb_metrics_prometheus(mk_request_t *request, void *data)
 
                 mk = sv.via.map.ptr[m].key;
                 mv = sv.via.map.ptr[m].val;
-
-                sds = flb_sds_cat(sds, "fluentbit_", 10);
-                sds = flb_sds_cat(sds, k.via.str.ptr, k.via.str.size);
-                sds = flb_sds_cat(sds, "_", 1);
-                sds = flb_sds_cat(sds, mk.via.str.ptr, mk.via.str.size);
-                sds = flb_sds_cat(sds, "_total{name=\"", 13);
-                sds = flb_sds_cat(sds, sk.via.str.ptr, sk.via.str.size);
-                sds = flb_sds_cat(sds, "\"} ", 3);
+                sds_metric = flb_sds_create_size(1024);
+                sds_metric = flb_sds_cat(sds_metric, "fluentbit_", 10);
+                sds_metric = flb_sds_cat(sds_metric, k.via.str.ptr, k.via.str.size);
+                sds_metric = flb_sds_cat(sds_metric, "_", 1);
+                sds_metric = flb_sds_cat(sds_metric, mk.via.str.ptr, mk.via.str.size);
+                sds_metric = flb_sds_cat(sds_metric, "_total{name=\"", 13);
+                sds_metric = flb_sds_cat(sds_metric, sk.via.str.ptr, sk.via.str.size);
+                sds_metric = flb_sds_cat(sds_metric, "\"} ", 3);
 
                 len = snprintf(tmp, sizeof(tmp) - 1,
                                "%" PRIu64 " ", mv.via.u64);
-                sds = flb_sds_cat(sds, tmp, len);
-                sds = flb_sds_cat(sds, time_str, time_len);
-                sds = flb_sds_cat(sds, "\n", 1);
+                sds_metric = flb_sds_cat(sds_metric, tmp, len);
+                sds_metric = flb_sds_cat(sds_metric, time_str, time_len);
+                sds_metric = flb_sds_cat(sds_metric, "\n", 1);
+                metrics_arr[index] = sds_metric;
+                index++;
             }
         }
     }
+
+    // Sort metrics in alphabetic order, so we can group them later.
```

Comment:
please use /* comment */ style as described in our contribution guidelines:

https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md